core/modules/replica/ConfigDatabase.h

```diff
@@ -67,8 +67,14 @@ class DatabaseInfo {
     /// @return The names of all tables.
     std::vector<std::string> tables() const;
 
-    std::string directorTable;      // The name of the Qserv "director" table if any.
-    std::string directorTableKey;   // The name of the primary key column in the "director" table.
+    std::string directorTable;          // The name of the Qserv "director" table if any.
+    std::map<std::string,               // The table name (partitioned tables only!).
+        std::string> directorTableKey;  // The name of the table's key representing object identifiers.
+                                        // NOTES: (1) In the "dependent" tables the key represents the FK
+                                        // associated with the corresponding PK of the "director" table.
+                                        // (2) The key is allowed to be empty for the "dependent" tables
+                                        // that don't have any objectId-based association with
+                                        // the "director" table.
```

Comment:
If there is no association with object table how the table can be dependent?

Reply:
This is an interesting question. I always thought that any partitioned table in Qserv is either the `director` or a `dependent` table of some `director`. The tables are partitioning is based either on the spatial coordinates or on the object identifiers leading to the corresponding entries of some other partitioned table (the `director` table). Does this mean I have a logic flow in the implementation of the Replication/Ingest system that (presently) allows a partitioned table that's is not explicitly presented to the system as the `director` not to have the objectId-based association to some  `director`. The current implementation still leaves this door "open". Should I fix this?

Reply:
No idea. But I think that partitioned table that is partitioned based on its own ra/dec is a director table and it is supposed to have unique/PK objectId. I'm not sure that we had any example of partitioned table without objectId. But I would ask @fritzm for definitive answer.